File: src/main/java/com/example/entities/db/Balance.java
package com.example.entities.db;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.validation.constraints.NotNull;

@Data
@Embeddable
public class Balance {

    @NotNull
    @Column(name = "BALANCE_AMOUNT")
    private int amount;

}

File: src/main/java/com/example/entities/db/Balance.kt
package com.example.entities.db

import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Generated
import org.hibernate.annotations.GenerationTime.INSERT
import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
data class Balance(
        @Generated(INSERT)
        @ColumnDefault("0")
        @Column(name = "BALANCE_AMOUNT", insertable = false)
        var amount: Int)